common/config/config.go

```diff
@@ -38,11 +38,13 @@ type Listener struct {
 	ProbeAddr            string `envconfig:"default=:8081,optional"`
 	SecureMetrics        bool   `envconfig:"default=true,optional"`
 	EnableHTTP2          bool   `envconfig:"default=false,optional"`
+	ApiExportWorkspace   string `envconfig:"default=:root,optional"`
+	ApiExportName        string `envconfig:"default=kubernetes.graphql.gateway,optional"`
 }
 
 // NewFromEnv creates a Gateway from environment values
-func NewFromEnv() (*Config, error) {
+func NewFromEnv() (Config, error) {
 	cfg := &Config{}
 	err := envconfig.Init(cfg)
-	return cfg, err
+	return *cfg, err
```

Comment:
I would suggest to write it this way:
```go
	cfg := Config{}
	err := envconfig.Init(&cfg)
	return cfg, err
```
No dereference and you explicitly show that `cfg` will be changed inside `envconfig.Init()`

Reply:
Yep, we can do this way as well :)